app/grandchallenge/evaluation/models.py

```diff
@@ -743,6 +784,37 @@ def _clean_external_evaluation(self):
                     "An external evaluation phase must have a parent phase."
                 )
 
+    def _clean_evaluation_requirements(self):
+        if self.selectable_gpu_types:
+            for option in self.selectable_gpu_types:
+                try:
+                    GPUTypeChoices[option]
+                except KeyError:
+                    raise ValidationError(
+                        f"{option!r} is not a valid option for selectable "
+                        f"gpu types. Options are {GPUTypeChoices.names}."
+                    )
+        if (
+            self.evaluation_requires_gpu_type
+            and self.evaluation_requires_gpu_type
+            not in self.get_selectable_gpu_types()
+        ):
+            raise ValidationError(
+                f"{self.evaluation_requires_gpu_type!r} is not a valid choice "
+                f"for Evaluation requires gpu type. Either change the choice or "
+                f"add it to the list of selectable gpu types."
+            )
+        if (
+            self.evaluation_requires_memory_gb
+            > self.maximum_settable_memory_gb
+        ):
+            raise ValidationError(
+                f"Ensure the value for Evaluation requires memory gb (currently "
+                f"{self.evaluation_requires_memory_gb}) is less than or equal "
+                f"to the maximum settable (currently "
+                f"{self.maximum_settable_memory_gb})."
+            )
```

Comment:
Aren't you mixing evaluation settings with algorithm settings here? This would keep a challenge admin from using more memory or a more powerful GPU for their evaluations than their participants algorithms, right?

Reply:
Yes that's correct. The 'maximum' and 'selectable' settings will be used for both algorithms and evaluations. I believe that was the intension, but I can easily add another setting for algorithms in the next PR.

Reply:
Ah ok, fine in that case.

Reply:
Actually - we may need to split this. I can think of a case where the challenge admin would no longer be able to update the phase form:

- They ask support for large instances, so we set 64GB evaluations
- They want to edit their phase form, validation would fail
- They don't want participants to select more than 32GB